src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -0,0 +1,219 @@
+package com.hms_networks.americas.sc.historicaldata;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * Class to manage queueing historical tag data and retrieving it in chunks based on a configurable
+ * time span of data.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.0
+ */
+public class HistoricalDataQueueManager {
+
+  /** Time span for fetching FIFO queue data. Default is 1 minute. */
+  private static long queueFifoTimeSpanMins = 1;
+
+  /**
+   * Get the current configured FIFO queue time span in milliseconds.
+   *
+   * @return FIFO queue time span in ms
+   */
+  private static synchronized long getQueueFifoTimeSpanMillis() {
+    return queueFifoTimeSpanMins
+        * HistoricalDataConstants.TIME_SECS_PER_MIN
+        * HistoricalDataConstants.TIME_MS_PER_SEC;
+  }
+
+  /**
+   * Get the current configured FIFO queue time span in minutes.
+   *
+   * @return FIFO queue time span in mins
+   */
+  public static synchronized long getQueueFifoTimeSpanMins() {
+    return queueFifoTimeSpanMins;
+  }
+
+  /**
+   * Set the FIFO queue time span in minutes.
+   *
+   * @param timeSpanMins
+   */
+  public static synchronized void setQueueFifoTimeSpanMins(long timeSpanMins) {
+    queueFifoTimeSpanMins = timeSpanMins;
+  }
+
+  /**
+   * Convert a long time pointer to format required for EDB calls.
+   *
+   * @param time long time pointer to format
+   * @return formatted time string for EBD calls
+   */
+  private static String convertToEBDTimeFormat(long time) {
+    return new SimpleDateFormat(HistoricalDataConstants.EBD_TIME_FORMAT).format(new Date(time));
+  }
+
+  /**
+   * Get the current long time pointer.
+   *
+   * @return current time as long
+   */
+  private static long getCurrentTime() {
+    return System.currentTimeMillis();
+  }
```

Comment:
Why does this exist?

Reply:
I made this separate because it is used in multiple locations, and in the previous implementation that Oliver created, it was a separate method that added an offset of time. This allows a quick modification to have the same offset functiionality

Reply:
at one point this was used to subtract a few seconds from the system time when getting the current time. That was to see if 1300+ tags was creating a lag in writing to the historical logs. I think this also had the timezone offset logic in here in the nordson project

Reply:
But that offset is not currently being used? If its not being used it probably shouldnt go in the repo.

Reply:
than aren't we stuck with UTC? I feel like the timezone code should probably get added in eventually

Reply:
removed (for now at least)